code/modules/mob/living/say.dm

```diff
@@ -134,8 +134,12 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	// AIs use inherent channels for the holopad. Most inherent channels
 	// ignore the language argument however.
 
-	if(handle_inherent_channels(message, message_mode, language)) //Hiveminds, binary chat & holopad.
-		return
+	var/key = lowertext(copytext(message, 2, 3))
+	for(var/_SM in subtypesof(/datum/saymode))
+		var/datum/saymode/SM = new _SM()
+		if((SM.key == key || SM.mode == message_mode) && SM.handle_message(src, message, language))

```

Comment:
\>Every time anyone says anything loop through every saymode type

Reply:
worst part: NOT CACHING THE STATELESS DATUMS.

Reply:
@RemieRichards what's a stateless datum?

Reply:
a datum that stores no state.
it comes from just the general concept of a stateless object in programming.

these saymodes always pass in their user/anything they need to use as arguments to procs, they don't ever store anything on themselves, meaning it's perfectly valid (and is in fact the BEST WAY TO DO THIS) to keep only 1 copy of each, shared between everything that speaks.

Reply:
@MoreRobustThanYou do saymodes ever get their vars changed if not why don't you just make a global list of them or something that's associative?

Reply:
and after i typed that i realized it's basically what remie said :P